evproc/processor.py

```diff
@@ -308,11 +311,16 @@ def process(self, ev):
                   ``StopProcessing`` exception raised by a processor.
         """
 
-        # Walk through all the processors and process the event
-        for proc in self._get_procs(ev):
-            try:
-                proc(ev)
-            except exc.StopProcessing as ex:
-                return ex.retval
+        event_deque = collections.deque([ev])
+        while event_deque:
+            ev = event_deque.popleft()
+            # Walk through all the processors and process the event
+            for proc in self._get_procs(ev):
+                try:
+                    additional_events = proc(ev)
+                except exc.StopProcessing as ex:
+                    return ex.retval
```

Comment:
Let's say you have two processors set up for event1, call them proc1 and proc2 (in that order).  Let's say proc1 fires off event2 and event3 by returning them as a list, and proc2 then raises StopProcessing.  The return here means that event2 and event3 will never be processed.  Is that what we should go for here?  I think the primary reasoning that I chose not to do the additional_events-type processing is because of that case, and I felt that this kind of architecture would be best served by an asynchronous event loop; evproc I designed as a synchronous event loop.  What do you think?